pkg/parser/parser.go

```diff
@@ -111,7 +110,9 @@ func (p *Parser) Parse(kubeFile []byte) (*v1.Image, error) {
 
 		switch layerType {
 		case Arg:
-			dispatchArg(layerValue, image)
+			if err := dispatchArg(layerValue, image); err != nil {
```

Comment:
Actually I do not agree the design of function `dispatchArg`.  This function includes two parts:
* dealing the raw data of `layerValue `;
* setting extracted data to `image`.
For decoupled code, we should separate them. Processing is processing, and setting and setting. And then, we could add more unit test for the code.